### PR TITLE
feature/books

### PR DESCRIPTION
**Pull request by TylerRockwell**
### Why?
https://github.com/smashingboxes/apprenticeship/tree/master/Curriculum/Backend/bookstore
Because users want to see details about the book they're looking to buy

### What Changed?
Added show action to BooksController
Made small tweaks to the site's layout

### CODE REVIEW

File: app/controllers/books_controller.rb [hunk 1]
```diff
@@ -1,5 +1,5 @@
 class BooksController < ApplicationController
-  before_action :set_book, only: [:edit, :update, :destroy]
+  before_action :set_book, only: [:edit, :update, :destroy, :show]
```
Comment (albaer) [inline app/controllers/books_controller.rb]: :smile:

File: spec/controllers/books_controller_spec.rb [hunk 2]
```diff
@@ -107,7 +107,16 @@
     let(:send_request){ put :update, params}
     it "updates the data in a Book object" do
       send_request
-      expect(assigns(:book).reload.title).to eq(valid_params[:title])
+      expect(assigns(:book).title).to eq(valid_params[:title])
```
Comment (albaer) [inline spec/controllers/books_controller_spec.rb]: Nice. Out of curiosity, why was the reload there and what happened to allow you to drop it?

File: spec/controllers/books_controller_spec.rb [hunk 3]
```diff
@@ -107,7 +107,16 @@
...
+    end
+  end
+
+  describe "PUT #update" do
+    let(:params){ { id: book.id } }
+    let(:send_request){ get :show, params}
```
Comment (albaer) [inline spec/controllers/books_controller_spec.rb]: You can change this to `let!(:send_request)` and then take it out of the `it` block. More info here: https://relishapp.com/rspec/rspec-core/v/2-6/docs/helper-methods/let-and-let

**Comment by albaer** [comment]
Nice work! :shipit: Take a look at `let` vs. `let!` to get familiar with who they work and when you might want to use one or the other.

File: spec/controllers/books_controller_spec.rb (hunk 2 above)
Comment (TylerRockwell) [inline spec/controllers/books_controller_spec.rb]: I thought I needed it, but I didn't. I just tried it without it to see if it would work, and it did.

Comment (albaer) [inline spec/controllers/books_controller_spec.rb]: :+1:

File: spec/controllers/books_controller_spec.rb (hunk 3 above)
Comment (TylerRockwell) [inline spec/controllers/books_controller_spec.rb]: Cool. I didn't know I could do that.

**Comment by TylerRockwell** [comment]
I will. Thanks for the review, @albaer !

**Pull request merged**